packages/editor/src/plugins/image-gallery/components/image-grid.tsx

```diff
@@ -1,58 +1,95 @@
+}: ImageGridProps) {
   return (
-    <div className="flex flex-wrap gap-4">
-      {images.map((image, index) => {
+    <div>
+      {images.map((leftImage, index) => {
         if (index % 2 !== 0) return null
 
-        // If the image is the last one in the array, render it as a single image
-        if (index + 1 === images.length) {
-          return (
-            <button
-              key={image.src}
-              className="mx-auto"
-              onMouseDown={() => onImageClick(index)}
-            >
-              <img
-                src={image.src}
-                alt={`Image ${image.src}`}
-                className="max-h-96"
-              />
-            </button>
-          )
-        }
-
-        const rowPercentages = getRowPercentages(image, images[index + 1])
+        const rightIndex = index + 1
+        const isLastImage = rightIndex === images.length
+        const rightImage = isLastImage ? undefined : images[rightIndex]
 
         return (
-          <Fragment key={image.src}>
-            <button
-              style={{ width: `calc(${rowPercentages.left}% - 0.5rem)` }}
-              onClick={() => onImageClick(index)}
-            >
-              <img src={image.src} alt={`Image ${image.src}`} />
-            </button>
-            <button
-              style={{ width: `calc(${rowPercentages.right}% - 0.5rem)` }}
-              onClick={() => onImageClick(index + 1)}
+          <div className="mb-4 flex gap-4" key={leftImage.src}>
+            <div
+              className={cn(wrapperClassNames, isLastImage && 'mx-auto')}
+              style={isLastImage ? {} : getFlex(leftImage)}
             >
-              <img
-                src={images[index + 1].src}
-                alt={`Image ${images[index + 1].src}`}
-              />
-            </button>
-          </Fragment>
+              <button onClick={() => onImageClick(index)}>
+                <img
+                  src={leftImage.src}
+                  // TODO: get actual alt text or fallback
+                  alt={`Image ${leftImage.src}`}
+                  className={cn(isLastImage && 'max-h-96')}
+                />
+              </button>
+              {extraChildren?.[index]}
+              {renderHoverOverlay(leftImage.caption, index)}
+            </div>
+            {rightImage ? (
+              <div className={wrapperClassNames} style={getFlex(rightImage)}>
+                <button onClick={() => onImageClick(rightIndex)}>
+                  <img src={rightImage.src} alt={`Image ${rightImage.src}`} />
+                </button>
+                {extraChildren?.[rightIndex]}
+                {renderHoverOverlay(rightImage.caption, rightIndex)}
+              </div>
+            ) : null}
+          </div>
         )
       })}
     </div>
   )
+
+  function renderHoverOverlay(caption: Descendant, index: number) {
+    return (
+      <div className={hoverOverlayClassNames}>
+        <StaticSlate element={caption} />
```

Comment:
The caption can include links. They have bad contrast (@GregorZupan) and they are not clickable right now (@hejtful)
<img width="126" alt="image" src="https://github.com/user-attachments/assets/a5471bb2-057a-4d7a-b319-c440914d3055">

Reply:
(code formatting and math formulas work okay)

Reply:
I'll fix the clickability

Reply:
Something like this might be enough:
```suggestion
    <div className="pointer-events-auto [&_a]:text-brand-100">
        <StaticSlate element={caption} />
    </div>
```

Reply:
Also added the color suggestion. @GregorZupan this is what it looks now, good enough?

<img width="250" alt="Screenshot 2024-08-29 at 16 30 05" src="https://github.com/user-attachments/assets/761cfd3f-9969-414c-9142-b6738a141c33">